service/tablet_allocator.cc

```diff
@@ -227,6 +264,75 @@ class load_balancer {
         }
     };
 
+    // Maximum tablet size works as the split threshold, whereas minimum size works as the
+    // merge threshold. To prevent oscillations between split and merge, hysteresis is applied
+    // to the decision.
+    // TODO: possibly make target_max_tablet_size a config to help writing tests that stresses resize path.
+    static const size_t target_max_tablet_size = 10UL * 1024 * 1024 * 1024;
+    static const size_t target_min_tablet_size = double(target_max_tablet_size / 2) * 0.6;
+
+    // This limits defines the maximum number of resize requests happening at a point in time.
+    // Due to power-of-two constraint, resize is always a cluster-wide operation meaning that
+    // a table will have all of its tablets resized, not only a subset. Therefore, it's important
+    // to set a limit on the operation, to avoid interfering with the foreground activity.
```

Comment:
Does this mean when we do split, the total number of tablets are always doubled. E.g, initial tablets = 8, after first split, we have 16 tablets, after second split we have 32 tablets. 

What if we have a "busy" tablet which gains much more data than the other tablets. Say 1 out of 8 tablets is having 8G of data, the target is 5GB per tablet, after we split, we have 16 tablets, 2 out of 16 are of size 4GB. (2 4GB tablets + 14 empty tablets).

Isn't it better to split the 8GB tablet into 8 tablets, not splitting the other 7 almost empty tablets? We still have 16 tablets after the split (8 1Gb tablets + 8 empty tablets)

Reply:
you're right but the whole tablet design is contrained by power of two. we plan to remove the constraint later, but not now.